### PR TITLE
Fixes shadow mesh being different than the lods.

### PR DESCRIPTION
**Pull request by fire**
The optimization of removing all the buffers except the vertex and lod indices should be enough without destroying the vertex order for lods.

### CODE REVIEW

File: editor/import/scene_importer_mesh.cpp [hunk 1]
```diff
@@ -323,71 +324,16 @@ void EditorSceneImporterMesh::create_shadow_mesh() {
...
-			for (int j = 0; j < index_count; j++) {
-				int index = index_rptr[j];
-				ERR_FAIL_INDEX(index, vertex_count);
-				index_wptr[j] = vertex_remap[index];
-			}
-
-			new_surface[RS::ARRAY_INDEX] = new_indices;
-
-			// Make sure the same LODs as the full version are used.
-			// This makes it more coherent between rendered model and its shadows.
```
Comment (Calinou) [inline editor/import/scene_importer_mesh.cpp:374]: Speaking of which, that comment reminds me of this proposal: https://github.com/godotengine/godot-proposals/issues/2600

**Comment by akien-mga** [comment]
You have two unrelated commits in this PR.

**Comment by fire** [comment]
@akien-mga I did a stacked patch approach. I removed it for clarity.

**Comment by reduz** [comment]
This PR makes no sense, if you do this then there is no point in having shadow meshes at all to begin with.

**Comment by reduz** [comment]
Welding is supposed to reduce the vertex buffer size considerably, specially in case of flat geometry. If you remove this, then the main argument for having shadow meshes is gone.

**Comment by fire** [comment]
We have to disable shadow meshes globally right now, please fix.

**Comment by fire** [comment]
JFonS will take a look at LOD and shadowmeshes when he is done with the subgizmos.

This pr is disabling shadow mesh which can be done easier in other ways.

**Pull request closed without merging**